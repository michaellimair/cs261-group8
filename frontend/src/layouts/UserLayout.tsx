import { FC } from 'react';
import {
  Box,
  useColorModeValue,
  Drawer,
  DrawerContent,
  useDisclosure,
} from '@chakra-ui/react';
import { Outlet } from 'react-router-dom';
import SidebarContent from 'components/Sidebar/SidebarLinkItem';
import SidebarMobileNav from 'components/Sidebar/SidebarMobileNav';

const SidebarWithHeader: FC = () => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <Box minH="100vh" bg={useColorModeValue('gray.100', 'gray.900')}>
      <SidebarContent
        onClose={onClose}
        display={{ base: 'none', md: 'block' }}
      />
      <Drawer
        autoFocus={false}
        isOpen={isOpen}
        placement="left"
        onClose={onClose}
        returnFocusOnClose={false}
        onOverlayClick={onClose}
        size="full"
      >
        <DrawerContent>
          <SidebarContent onClose={onClose} />
        </DrawerContent>
      </Drawer>
      <SidebarMobileNav onOpen={onOpen} />
      <Box ml={{ base: 0, md: 60 }} p="4">
        <Outlet />
      </Box>
    </Box>
  );
};

export default SidebarWithHeader;
