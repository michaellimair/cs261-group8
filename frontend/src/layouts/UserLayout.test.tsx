import { render } from '@testing-library/react';
import UserLayout from './UserLayout';

describe('UserLayout', () => {
  it('renders properly', () => {
    const result = render(<UserLayout />);
    expect(result).toMatchSnapshot();
  });
});
