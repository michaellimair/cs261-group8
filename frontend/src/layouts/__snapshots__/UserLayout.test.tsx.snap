// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLayout renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-2k1zm6"
      >
        <div
          class="css-rf1m55"
        >
          <div
            class="css-1bxj9j3"
          >
            <p
              class="chakra-text css-1gfp23t"
            >
              Logo
            </p>
            <button
              aria-label="Close"
              class="css-33zxup"
              data-testid="closeButton"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
          <a
            class="chakra-link css-jgup0q"
            href="/"
            style="text-decoration: none;"
          >
            <div
              class="css-cpuzs9"
              role="group"
            >
              <svg
                class="chakra-icon css-1b1fm3x"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                />
                <polyline
                  points="9 22 9 12 15 12 15 22"
                />
              </svg>
              Home
            </div>
          </a>
          <a
            class="chakra-link css-jgup0q"
            href="/"
            style="text-decoration: none;"
          >
            <div
              class="css-cpuzs9"
              role="group"
            >
              <svg
                class="chakra-icon css-1b1fm3x"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="23 6 13.5 15.5 8.5 10.5 1 18"
                />
                <polyline
                  points="17 6 23 6 23 12"
                />
              </svg>
              Trending
            </div>
          </a>
          <a
            class="chakra-link css-jgup0q"
            href="/"
            style="text-decoration: none;"
          >
            <div
              class="css-cpuzs9"
              role="group"
            >
              <svg
                class="chakra-icon css-1b1fm3x"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polygon
                  points="16.24 7.76 14.12 14.12 7.76 16.24 9.88 9.88 16.24 7.76"
                />
              </svg>
              Explore
            </div>
          </a>
          <a
            class="chakra-link css-jgup0q"
            href="/"
            style="text-decoration: none;"
          >
            <div
              class="css-cpuzs9"
              role="group"
            >
              <svg
                class="chakra-icon css-1b1fm3x"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              Favourites
            </div>
          </a>
          <a
            class="chakra-link css-jgup0q"
            href="/"
            style="text-decoration: none;"
          >
            <div
              class="css-cpuzs9"
              role="group"
            >
              <svg
                class="chakra-icon css-1b1fm3x"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
                <path
                  d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
                />
              </svg>
              Settings
            </div>
          </a>
        </div>
        <div
          class="css-g8fv48"
        >
          <button
            aria-label="open menu"
            class="chakra-button css-2mi4sz"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="3"
                x2="21"
                y1="12"
                y2="12"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <line
                x1="3"
                x2="21"
                y1="18"
                y2="18"
              />
            </svg>
          </button>
          <p
            class="chakra-text css-1ri2w28"
          >
            Logo
          </p>
          <div
            class="chakra-stack css-1qgtx64"
          >
            <button
              aria-label="open menu"
              class="chakra-button css-198bfjk"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 8A6 6 0 0 0 6 8c0 7-3 9-3 9h18s-3-2-3-9"
                />
                <path
                  d="M13.73 21a2 2 0 0 1-3.46 0"
                />
              </svg>
            </button>
            <div
              class="css-70qvj9"
            >
              <button
                aria-controls="menu-list-7"
                aria-expanded="false"
                aria-haspopup="menu"
                class="chakra-menu__menu-button css-10lilp"
                id="menu-button-7"
              >
                <span
                  class="css-xl71ch"
                >
                  <div
                    class="chakra-stack css-84zodg"
                  >
                    <span
                      class="chakra-avatar css-1ruhx4w"
                    >
                      <svg
                        aria-label=" avatar"
                        class="chakra-avatar__svg css-16ite8i"
                        role="img"
                        viewBox="0 0 128 128"
                      >
                        <path
                          d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                          fill="currentColor"
                        />
                        <path
                          d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                    <div
                      class="chakra-stack css-q43qan"
                    >
                      <p
                        class="chakra-text css-g5hezf"
                      >
                        Justina Clark
                      </p>
                      <p
                        class="chakra-text css-65bt76"
                      >
                        Admin
                      </p>
                    </div>
                    <div
                      class="css-1t6irk0"
                    >
                      <svg
                        fill="none"
                        height="1em"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polyline
                          points="6 9 12 15 18 9"
                        />
                      </svg>
                    </div>
                  </div>
                </span>
              </button>
              <div
                class="css-0"
                style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
              >
                <div
                  aria-orientation="vertical"
                  class="chakra-menu__menu-list css-w8sf4k"
                  id="menu-list-7"
                  role="menu"
                  style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
                  tabindex="-1"
                >
                  <button
                    class="chakra-menu__menuitem css-1k756t9"
                    data-index="0"
                    id="menu-list-7-menuitem-2"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    Profile
                  </button>
                  <button
                    class="chakra-menu__menuitem css-1k756t9"
                    data-index="1"
                    id="menu-list-7-menuitem-3"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    Settings
                  </button>
                  <button
                    class="chakra-menu__menuitem css-1k756t9"
                    data-index="2"
                    id="menu-list-7-menuitem-4"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    Billing
                  </button>
                  <hr
                    aria-orientation="horizontal"
                    class="chakra-menu__divider css-0"
                    role="separator"
                  />
                  <button
                    class="chakra-menu__menuitem css-1k756t9"
                    data-index="3"
                    id="menu-list-7-menuitem-5"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    Sign out
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1fr3uq1"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-2k1zm6"
    >
      <div
        class="css-rf1m55"
      >
        <div
          class="css-1bxj9j3"
        >
          <p
            class="chakra-text css-1gfp23t"
          >
            Logo
          </p>
          <button
            aria-label="Close"
            class="css-33zxup"
            data-testid="closeButton"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <a
          class="chakra-link css-jgup0q"
          href="/"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
              />
              <polyline
                points="9 22 9 12 15 12 15 22"
              />
            </svg>
            Home
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="23 6 13.5 15.5 8.5 10.5 1 18"
              />
              <polyline
                points="17 6 23 6 23 12"
              />
            </svg>
            Trending
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <polygon
                points="16.24 7.76 14.12 14.12 7.76 16.24 9.88 9.88 16.24 7.76"
              />
            </svg>
            Explore
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            Favourites
          </div>
        </a>
        <a
          class="chakra-link css-jgup0q"
          href="/"
          style="text-decoration: none;"
        >
          <div
            class="css-cpuzs9"
            role="group"
          >
            <svg
              class="chakra-icon css-1b1fm3x"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="3"
              />
              <path
                d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
              />
            </svg>
            Settings
          </div>
        </a>
      </div>
      <div
        class="css-g8fv48"
      >
        <button
          aria-label="open menu"
          class="chakra-button css-2mi4sz"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="none"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="3"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              x1="3"
              x2="21"
              y1="18"
              y2="18"
            />
          </svg>
        </button>
        <p
          class="chakra-text css-1ri2w28"
        >
          Logo
        </p>
        <div
          class="chakra-stack css-1qgtx64"
        >
          <button
            aria-label="open menu"
            class="chakra-button css-198bfjk"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 8A6 6 0 0 0 6 8c0 7-3 9-3 9h18s-3-2-3-9"
              />
              <path
                d="M13.73 21a2 2 0 0 1-3.46 0"
              />
            </svg>
          </button>
          <div
            class="css-70qvj9"
          >
            <button
              aria-controls="menu-list-7"
              aria-expanded="false"
              aria-haspopup="menu"
              class="chakra-menu__menu-button css-10lilp"
              id="menu-button-7"
            >
              <span
                class="css-xl71ch"
              >
                <div
                  class="chakra-stack css-84zodg"
                >
                  <span
                    class="chakra-avatar css-1ruhx4w"
                  >
                    <svg
                      aria-label=" avatar"
                      class="chakra-avatar__svg css-16ite8i"
                      role="img"
                      viewBox="0 0 128 128"
                    >
                      <path
                        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                        fill="currentColor"
                      />
                      <path
                        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                  <div
                    class="chakra-stack css-q43qan"
                  >
                    <p
                      class="chakra-text css-g5hezf"
                    >
                      Justina Clark
                    </p>
                    <p
                      class="chakra-text css-65bt76"
                    >
                      Admin
                    </p>
                  </div>
                  <div
                    class="css-1t6irk0"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <polyline
                        points="6 9 12 15 18 9"
                      />
                    </svg>
                  </div>
                </div>
              </span>
            </button>
            <div
              class="css-0"
              style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
            >
              <div
                aria-orientation="vertical"
                class="chakra-menu__menu-list css-w8sf4k"
                id="menu-list-7"
                role="menu"
                style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
                tabindex="-1"
              >
                <button
                  class="chakra-menu__menuitem css-1k756t9"
                  data-index="0"
                  id="menu-list-7-menuitem-2"
                  role="menuitem"
                  tabindex="-1"
                  type="button"
                >
                  Profile
                </button>
                <button
                  class="chakra-menu__menuitem css-1k756t9"
                  data-index="1"
                  id="menu-list-7-menuitem-3"
                  role="menuitem"
                  tabindex="-1"
                  type="button"
                >
                  Settings
                </button>
                <button
                  class="chakra-menu__menuitem css-1k756t9"
                  data-index="2"
                  id="menu-list-7-menuitem-4"
                  role="menuitem"
                  tabindex="-1"
                  type="button"
                >
                  Billing
                </button>
                <hr
                  aria-orientation="horizontal"
                  class="chakra-menu__divider css-0"
                  role="separator"
                />
                <button
                  class="chakra-menu__menuitem css-1k756t9"
                  data-index="3"
                  id="menu-list-7-menuitem-5"
                  role="menuitem"
                  tabindex="-1"
                  type="button"
                >
                  Sign out
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1fr3uq1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
